ledger-core/v2/conveyor/conveyor.go

```diff
@@ -107,6 +107,14 @@ func (p *PulseConveyor) AddDependency(v interface{}) {
 	p.slotMachine.AddDependency(v)
 }
 
+func (p *PulseConveyor) AddDependencyByName(id string, v interface{}) {
+	p.slotMachine.AddDependencyByName(id, v)
+}
+
+func (p *PulseConveyor) AddDependencyByType(vt interface{}, v interface{}) {
```

Comment:
from what I see this is not required or at least wrongly named. In every use you pass pointer to a value as first argument and value as second argument. Meaning of this is not clear to me.

Reply:
When we use SlotMachine.AddDependency with interface go type system return name of struct but not an interface. But when DI try inject interface to state machine it can't find name (name of struct != name of interface). @cyraxred say what need to create new function for that case. I agree that name of function must be better but i don't know how to name it